### PR TITLE
Fix to allow UM and PINNED memory to be active simultaneously

### PR DESCRIPTION
**Pull request by liu15**

### CODE REVIEW

File: src/chai/ArrayManager.cpp [hunk 1]
```diff
@@ -326,7 +327,7 @@ void ArrayManager::free(PointerRecord* pointer_record, ExecutionSpace spaceToFre
               if (space_ptr == pointer_record->m_pointers[space_t])
                 pointer_record->m_pointers[space_t] = nullptr;
             }
-          } else {
+          } else
```
Comment (adayton1) [inline src/chai/ArrayManager.cpp:330]: I think this `else` without an opening curly brace is problematic.

Comment (robinson96) [reply src/chai/ArrayManager.cpp:330]: I think so too - but also concerned that testing isn't catching a potential mismatch bracket issue?

Comment (rhornung67) [reply src/chai/ArrayManager.cpp:330]: IMO, there should also be curly braces around line 328 above as well for readability, at least.

Comment (adayton1) [reply src/chai/ArrayManager.cpp:330]: So without the brace, was it just executing the callback line and missing the deallocate?

Comment (liu15) [reply src/chai/ArrayManager.cpp:330]: Good catch.  Without the brace it is valid code, just not what I wanted, I think.  Only the first line will be in the conditional clause and the rest will be unconditional.

**Review by adayton1 — APPROVED** [review]
LGTM

File: src/chai/ArrayManager.cpp (hunk 1 above)
Comment (liu15) [reply src/chai/ArrayManager.cpp:332]: without the brace, the deallocation would have been called twice in the other cases (other than the else).

Comment (rhornung67) [reply src/chai/ArrayManager.cpp:332]: @liu15 thanks for the curly brace additions. I'm not as sad anymore :-)

**Comment by davidbeckingsale** [comment]
LGTM

**Review by adayton1 — APPROVED** [review]
LGTM

**Review by adayton1 — APPROVED** [review]
LGTM

**Review by robinson96 — APPROVED** [review]
LGTM

**Pull request merged**